pkg/vsphere/disk/disk_manager.go

```diff
@@ -142,6 +142,21 @@ func (m *Manager) toSpec(config *VirtualDiskConfig) *types.VirtualDisk {
 func (m *Manager) CreateAndAttach(op trace.Operation, config *VirtualDiskConfig) (*VirtualDisk, error) {
 	defer trace.End(trace.Begin(config.DatastoreURI.String()))
 
+	// reuse the disk if it already exists and is attached
+	if _, err := findDiskByFilename(op, m.vm, config.DatastoreURI.String()); err == nil {
+		// if the disk is attached, it should be in the disk cache
+		d, err := m.Get(op, config)
+		if err != nil {
+			return nil, err
+		}
+		d.l.Lock()
```

Comment:
The locks aren't used specifically to bump reference counters, only in this case (if I recall correctly). They are used elsewhere while creating, attaching, etc.